client/src/models/Task.js

```diff
@@ -1,9 +1,25 @@
 import { Settings } from "api"
-import Model, { createYupObjectShape, yupDate } from "components/Model"
+import Model, {
+  createYupObjectShape,
+  NOTE_TYPE,
+  yupDate
+} from "components/Model"
+import _isEmpty from "lodash/isEmpty"
+import { Report } from "models"
 import TASKS_ICON from "resources/tasks.png"
 import utils from "utils"
 import * as yup from "yup"
 
+function createTaskAssessmentSchema(customFieldsConfig) {
+  const taskAssessmentSchemaShape = createYupObjectShape(
+    customFieldsConfig,
+    "entityAssessment"
```

Comment:
Maybe also `export const ENTITY_ASSESSMENT_FIELD = "entityAssessment"` seeing as it's used in several places?

Reply:
I will do that as part of the branch related to #2954 , the code is there a bit different and more generic.